tests/cli/test_remote.py

```diff
@@ -1,7 +1,731 @@
+        assert home_dir.startswith("/home/mila/")
+    else:
+        mock_connection.run.assert_called_once_with(
+            "echo $HOME", asynchronous=False, hide=True, warn=False, out_stream=None
+        )
+        mock_connection.local.assert_not_called()
+
+
+@pytest.mark.skipif(
+    "-s" not in sys.argv, reason="TODO: Seems to require the -s option?!"
+)
+@dont_run_for_real
+@disable_internet_access
+def test_persist(mock_connection: Connection, host: str, capsys: pytest.CaptureFixture):
+    r = Remote(host, connection=mock_connection)
+    r.persist()
+    assert (
+        "Warning: --persist does not work with --node or --job"
+        in capsys.readouterr().out
+    )
+
+
+@pytest.fixture
+def remote(mock_connection: Connection, host: str) -> Remote:
+    return Remote(hostname=host, connection=mock_connection)
+
+
+@dont_run_for_real
+@disable_internet_access
+def test_ensure_allocation(remote: Remote):
+    assert remote.ensure_allocation() == ({"node_name": remote.hostname}, None)
+
+
+def some_transform(x: str) -> str:
+    return f"echo Hello && {x}"
+
+
+def some_other_transform(x: str) -> str:
+    return f"echo 'this is printed after the command' && {x}"
+
+
+class TestSlurmRemote:
+    @dont_run_for_real
+    @pytest.mark.parametrize("persist", [True, False])
+    def test_init_(self, mock_connection: Connection, persist: bool):
+        alloc = ["--time=00:01:00"]
+        transforms = [some_transform]
+        remote = SlurmRemote(
+            mock_connection, alloc=alloc, transforms=transforms, persist=persist
+        )
+        # TODO: This kind of test feels a bit dumb.
+        assert remote.connection is mock_connection
+        assert remote._persist == persist
+        assert remote.transforms == [
+            *transforms,
+            remote.srun_transform_persist if persist else remote.srun_transform,
+        ]
+
+    @can_run_for_real
+    def test_srun_transform(self, mock_connection: Connection):
+        alloc = ["--time=00:01:00"]
+        transforms = [some_transform]
+        persist: bool = False
+        remote = SlurmRemote(
+            mock_connection, alloc=alloc, transforms=transforms, persist=persist
+        )
+        # Transforms aren't used here. Seems a bit weird for this to be a public method
+        # then, no?
+        assert remote.srun_transform("bob") == "srun --time=00:01:00 bash -c bob"
+
+    @dont_run_for_real
+    @pytest.mark.skip(reason="Seems a bit hard to test for what it's worth..")
+    def test_srun_transform_persist(self, mock_connection: Connection):
+        alloc = ["--time=00:01:00"]
+        transforms = [some_transform]
+        persist: bool = False
+        remote = SlurmRemote(
+            mock_connection, alloc=alloc, transforms=transforms, persist=persist
+        )
+        output_file = "<some_file>"
+        assert (
+            remote.srun_transform_persist("bob")
+            == f"bob; touch {output_file}; tail -n +1 -f {output_file}"
+        )
```

Comment:
Same, I think we could mock and use a `FileRegressionFixture` or something like that

Reply:
Yeah, I think I see what you mean, you want to avoid the hard-coded portion of the test and move that to a regression file instead?

Reply:
Yes that's what I had in mind